src/common/util/feature-flags.js

```diff
@@ -24,21 +23,17 @@ export function activateFeatures (flags, agentIdentifier) {
   if (sentIds.has(agentIdentifier)) return
 
   Object.entries(flags).forEach(([flag, num]) => {
-    if (bucketMap[flag]) {
-      bucketMap[flag].forEach(feat => {
-        if (!num) handle('block-' + flag, [], undefined, feat, sharedEE)
-        else handle('feat-' + flag, [], undefined, feat, sharedEE)
-        handle('rumresp-' + flag, [Boolean(num)], undefined, feat, sharedEE) // this is a duplicate of feat-/block- but makes awaiting for 1 event easier than 2
-      })
-    } else if (num) handle('feat-' + flag, [], undefined, undefined, sharedEE) // not sure what other flags are overlooked, but there's a test for ones not in the map --
+    if (!num) sharedEE.emit('block-' + flag, [])
+    else sharedEE.emit('feat-' + flag, [])
+    sharedEE.emit('rumresp-' + flag, [num]) // this is a duplicate of feat-/block- but makes awaiting for 1 event easier than 2
```

Comment:
using handle here creates an unneeded race condition between emitting to features to enable them and being able to drain.  All features should already be present on the page before this emits. This is the pattern that was used in <=v1216 for many years, and is simpler to rely on.  This allows us to no longer need the middleware buffer in ST for the re-write.